### PR TITLE
Get response content-type

### PR DESCRIPTION
**Pull request by seans3**
* Result getter for response `Content-Type` header
* Used for accept/content-type negotiation

Example
```
var apiGroupList metav1.APIGroupList
var responseContentType string
err = d.restClient.Get()
  .AbsPath("/apis")
  .SetHeader("Accept", orderedContentTypes)
  .Do(context.TODO())
  .ContentType(&responseContentType)
  .Into(&apiGroupList)
```

/kind feature

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/rest/request.go [hunk 1]
```diff
@@ -1288,6 +1288,13 @@ func (r Result) StatusCode(statusCode *int) Result {
 	return r
 }
 
+// ContentType returns the "Content-Type" response header into the passed
+// string, returning the Result for possible chaining.
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/request.go:1292]: does this need a similar disclaimer as StatusCode that it is only valid if no error was returned?

Comment (seans3) [reply staging/src/k8s.io/client-go/rest/request.go]: Good catch. Yes--similar disclaimer needed and added.

**Pull request merged**